Clean travis.yml and docs (#5716)
puppeteer/puppeteer: docs/troubleshooting.md

> 👋 We run our tests for Puppeteer on Travis CI - see our [`.travis.yml`](https://github.com/puppeteer/puppeteer/blob/master/.travis.yml) for reference.

Tips-n-tricks:
- [xvfb](https://en.wikipedia.org/wiki/Xvfb) service should be launched in order to run Chromium in non-headless mode
- `node_modules` is cached by default

> - os linux with Xenial dist in Travis by default

Reviewer: ```suggestion
- Runs on Xenial Linux on Travis by default
```
